i6_models/parts/factored_hybrid/util.py

```diff
@@ -0,0 +1,63 @@
+from enum import Enum
+from typing import Callable, Union
+
+from torch import nn
+
+
+class PhonemeStateClassV1(Enum):
+    """Phoneme state class augmentation selector"""
+
+    none = 1
+    word_end = 2
+    boundary = 4
+
+    def factor(self):
+        return self.value
+
+
+def get_center_dim(
+    n_contexts: int,
+    num_hmm_states_per_phone: int,
+    ph_class: Union[int, PhonemeStateClassV1],
+) -> int:
+    """
+    :return: number of center phonemes given the augmentation values
+    """
+
+    factor = ph_class.factor() if isinstance(ph_class, PhonemeStateClassV1) else ph_class
+    return n_contexts * num_hmm_states_per_phone * factor
+
+
+def get_mlp(
```

Comment:
There is also https://github.com/rwth-i6/i6_models/blob/main/i6_models/parts/ffnn.py that one could reuse. But some linear layers seems simple enough to me that I do not care about code duplication here.

Reply:
That one does not work with `nn.Sequential`, which is why I don't use it here.

Reply:
offline discussion: this does not work for nn.Sequential